### ISSUE TITLE
How i can change the background img of public html ?

### ISSUE THREAD

**Issue by m333w** [AzuraCast/AzuraCast#724]
Hi all

I woud like to change de backgroup of the public html, in "Custom branding i have add 
`background-image: url("https://url.of.the.img/");`

But nothing have change, someone can help me ?

Best regards.

**Comment by djmuratb** [comment]
/var/azuracast/www/web/static/img/hexbg.png  &  hexbg_dark.png

**Comment by m333w** [comment]
Thx for reply @djmuratb, so i just have to replace hexbg.png with my img with the same name ? You sure is a good practice ?

**Comment by SlvrEagle23** [comment]
@m333w Just to clarify, you can indeed use the "Custom CSS for Public Pages" section to customize the background image, you just need to specify it with the full CSS selector, like so:

```css
.four-zero, .login-content, .public-page-content {
    background: #edecec url('https://picsum.photos/1920/1080?image=39&blur') center center;
}
```

This will allow you to replace the image without needing to make any material changes to the AzuraCast codebase itself.

**Comment by m333w** [comment]
Hi @SlvrEagle23,

Thx very mutch for your reply and this "beter" solution. I gona try in few hours actualy i am not at home.
I keep you noticed about this issue :) 

Best regards.

**Comment by ErnestPH** [comment]
> @m333w Just to clarify, you can indeed use the "Custom CSS for Public Pages" section to customize the background image, you just need to specify it with the full CSS selector, like so:
> 
> ```css
> .four-zero, .login-content, .public-page-content {
>     background: #edecec url('https://picsum.photos/1920/1080?image=39&blur') center center;
> }
> ```
> 
> This will allow you to replace the image without needing to make any material changes to the AzuraCast codebase itself.

After the latest update, the CSS method is not working anymore

Thanks!

**Comment by SlvrEagle23** [comment]
@m333w @ErnestPH @djmuratb Just as an update, as a result of the design update where we rolled out a newer version of our underlying UI framework, the CSS selector for the background image on the public pages has changed.

You should now be able to use:

```css
body.page-minimal {
    background: #edecec url('https://picsum.photos/1920/1080?image=39&blur') center center;
}
```

...to set the page's background. Of course, feel free to replace the URL and hex color in the CSS above as needed.

**Comment by ErnestPH** [comment]
Yes of course, I forgot about that. haha What was I thinking? :) Thanks again!

**Comment by m333w** [comment]
@SlvrEagle23 Thank you very mutch :)

**Comment by isAAAc** [comment]
there is my CSS to change the background, responsive by design ;)
you can see the result on https://stream.krashboyz.org/public/krashboyz_bordel_klub 

Enjoy

```
body.page-minimal {
    background: #edecec url('https://pic.krashboyz.org/lutim/g8MWeW3X/0U5W3Nte.jpg') center center;
    background-repeat: no-repeat;
    background-color: black;
    background-size: contain;
}
```

**Comment by janosnet** [comment]
> there is my CSS to change the background, responsive by design ;)
> you can see the result on https://stream.krashboyz.org/public/krashboyz_bordel_klub
> 
> Enjoy
> 
> ```
> body.page-minimal {
>     background: #edecec url('https://pic.krashboyz.org/lutim/g8MWeW3X/0U5W3Nte.jpg') center center;
>     background-repeat: no-repeat;
>     background-color: black;
>     background-size: contain;
> }
> ```

thanks dear

**Comment by kingbalo** [comment]
> 
> 
> @m333w @ErnestPH @djmuratb Just as an update, as a result of the design update where we rolled out a newer version of our underlying UI framework, the CSS selector for the background image on the public pages has changed.
> 
> You should now be able to use:
> 
> ```css
> body.page-minimal {
>     background: #edecec url('https://picsum.photos/1920/1080?image=39&blur') center center;
> }
> ```
> 
> ...to set the page's background. Of course, feel free to replace the URL and hex color in the CSS above as needed.


How can I reference a background image on the local file system in for example /var/azuracast/background.jpg  

I tried this: 

body.page-minimal {
    background: #edecec url("var/azuracast/abfm-bg.jpg") center center;
}

Doesn't work.

**Comment by SlvrEagle23** [comment]
@kingbalo We've since updated the CSS further; try this:

```css
[data-theme] body.page-minimal {
    background: #edecec url("/var/azuracast/background.jpg") center center;
}
```

**Comment by kingbalo** [comment]
> 
> 
> @kingbalo We've since updated the CSS further; try this:
> 
> ```css
> [data-theme] body.page-minimal {
>     background: #edecec url("/var/azuracast/background.jpg") center center;
> }
> ```


Did Not Work. Let me elaborate.

I'm running Azuracast with docker and I'm trying to reference an image file on the host system to use as the background for the public pages.

Here's what i did:

1. created a directory /var/azuracast/img which contained the background image.
2. created the docker-compose.override.yml file in /var/azuracast to bind mount the image directory on the host to a custom directory in the web service container. see code below:

```
version: '2.2'
services:
    web:
        volumes:
            - /var/azuracast/img:/var/azuracast/www/web/static/img/custom
```

3. I restarted Azuracast with `docker-compose down && docker-compose up -d`
4. I used [portainer ](https://www.portainer.io/)to access the console of the azuracast_web container and verified that the bind mount was successful.
5. Logged in to Azuracast and added the custom CSS as you indicated, referencing the image file with the absolute path in the azuracast_web container as below:

```
[data-theme] body.page-minimal {
    background: #edecec url("/var/azuracast/www/web/static/img/custom/background.jpg") center center;
}
```
6. Did Not Work. Still loads the default background image.
7. Please advice 😁

**Comment by SlvrEagle23** [comment]
@kingbalo The URL should be relative to the web root, not the actual path on disk, so it should be something like "/static/img/custom/background.jpg" instead of the full path.

**Comment by kingbalo** [comment]
@SlvrEagle23 Thank you very much 👍🏽.

It finally worked. I had to use the old CSS method though I'm on the latest stable v0.12.4. 
The new CSS method fails and still loads the default background.

Old CSS method (worked):
```
body.page-minimal {
    background: #edecec url("/static/img/custom/background.jpg") center center;
}
```
New CSS method (failed):
```
[data-theme] body.page-minimal {
    background: #edecec url("/static/img/custom/background.jpg") center center;
}
```

**Comment by ErnestPH** [comment]
@kingbalo Just letting you know that new css method should work...

```
[data-theme=dark] body.page-minimal {
   background: #edecec url("https://domainname/path-to-image") center center !important;
}
```
